core/src/sumeragi/mod.rs

```diff
@@ -93,25 +104,71 @@ impl Sumeragi {
                 telemetry_started,
                 transaction_limits: configuration.transaction_limits,
                 transaction_validator,
-                genesis_network,
                 queue,
                 broker,
                 kura,
-                // network,
+                network,
                 actor_channel_capacity: configuration.actor_channel_capacity,
                 fault_injection: PhantomData,
                 gossip_batch_size: configuration.gossip_batch_size,
                 gossip_period: Duration::from_millis(configuration.gossip_period_ms),
+
+                atomic_sumeragi_state: Mutex::new(atomic_sumeragi_state),
+                current_online_peers_by_public_key: Mutex::new(Vec::new()),
+                incomming_message_sender: Mutex::new(incomming_message_sender),
+                incomming_message_receiver: Mutex::new(incomming_message_receiver),
             },
         })
     }
 
+    pub fn update_metrics(&self, network: Addr<IrohaNetwork>) -> Result<()> {
+        use eyre::WrapErr;
+        use thiserror::Error;
+        let online_peers = self
+            .internal
+            .current_online_peers_by_public_key
+            .lock()
+            .unwrap();
```

Comment:
Consider returning as a recoverable error.

Reply:
good idea

Reply:
Turns out, actually not a good idea. Mutex poison errors can't be passed between threads safely.